### ISSUE TITLE
Deleting Assets Prevents Browsing Assets Untill Cache Clear

### ISSUE THREAD

**Issue by jcohlmeyer** [statamic/v2-hub#946]
### Expected behaviour
* After deleting an asset I should be able to brows assets without running `php please clear:cache`

### Actual behaviour
* After deleting an asset I have to run `php please clear:cache` in order to browse files. I see a 500 internal server error for the network request: `/cp/assets/browse`

### Steps to reproduce
1.  Upload an asset with cp
2.  Delete an asset with cp
3.  Try to browse that folder again with Add Asset or the Assets browser

### Server configuration
**Operating system:** CENTOS 6.8

**Web server:** Apache 2.4 

**PHP version:** PHP 5.6 

**Statamic version:** 2.1.9

**Updated from an older Statamic or fresh install:** Started at 2. something

**List of installed addons:** none

### Logs
```
Stack trace:
#0 /home/trinitylife/public_html/statamic/core/CP/FieldtypeFactory.php(21): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(4096, 'Argument 2 pass...', '/home/trinityli...', 21, Array)
#1 /home/trinitylife/public_html/statamic/bundles/Grid/GridFieldtype.php(65): Statamic\CP\FieldtypeFactory::create('text', NULL)
#2 /home/trinitylife/public_html/statamic/bundles/Grid/GridFieldtype.php(55): Statamic\Addons\Grid\GridFieldtype->processField(Array, NULL)
#3 /home/trinitylife/public_html/statamic/bundles/Grid/GridFieldtype.php(42): Statamic\Addons\Grid\GridFieldtype->processRow(Array)
#4 /home/trinitylife/public_html/statamic/bundles/Grid/GridFieldtype.php(34): Statamic\Addons\Grid\GridFieldtype->performProcess(Array)
#5 /home/trinitylife/public_html/statamic/core/CP/Publish/Publisher.php(200): Statamic\Addons\Grid\GridFieldtype->process(Array)
#6 /home/trinitylife/public_html/statamic/core/CP/Publish/Publisher.php(86): Statamic\CP\Publish\Publisher->processFields()
#7 /home/trinitylife/public_html/statamic/core/CP/Publish/EntryPublisher.php(25): Statamic\CP\Publish\Publisher->publish()
#8 /home/trinitylife/public_html/statamic/core/Http/Controllers/PublishController.php(433): Statamic\CP\Publish\EntryPublisher->publish()
#9 [internal function]: Statamic\Http\Controllers\PublishController->save()
#10 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(256): call_user_func_array(Array, Array)
#11 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(164): Illuminate\Routing\Controller->callAction('save', Array)
#12 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(112): Illuminate\Routing\ControllerDispatcher->call(Object(Statamic\Http\Controllers\PublishController), Object(Illuminate\Routing\Route), 'save')
#13 [internal function]: Illuminate\Routing\ControllerDispatcher->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#14 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(139): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#15 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#16 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#17 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(114): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#18 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(69): Illuminate\Routing\ControllerDispatcher->callWithinStack(Object(Statamic\Http\Controllers\PublishController), Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'save')
#19 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/Route.php(203): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'Statamic\\Http\\C...', 'save')
#20 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/Route.php(134): Illuminate\Routing\Route->runWithCustomDispatcher(Object(Illuminate\Http\Request))
#21 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/Router.php(708): Illuminate\Routing\Route->run(Object(Illuminate\Http\Request))
#22 [internal function]: Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#23 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(139): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#24 /home/trinitylife/public_html/statamic/core/Http/Middleware/CP/Authenticate.php(62): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#25 [internal function]: Statamic\Http\Middleware\CP\Authenticate->handle(Object(Illuminate\Http\Request), Object(Closure))
#26 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#27 /home/trinitylife/public_html/statamic/core/Http/Middleware/Outpost.php(52): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#28 [internal function]: Statamic\Http\Middleware\Outpost->handle(Object(Illuminate\Http\Request), Object(Closure))
#29 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#30 /home/trinitylife/public_html/statamic/core/Http/Middleware/CP/DefaultLocale.php(28): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#31 [internal function]: Statamic\Http\Middleware\CP\DefaultLocale->handle(Object(Illuminate\Http\Request), Object(Closure))
#32 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#33 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#34 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#35 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/Router.php(710): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#36 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/Router.php(675): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#37 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Routing/Router.php(635): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#38 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(236): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#39 [internal function]: Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#40 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(139): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#41 /home/trinitylife/public_html/statamic/core/Http/Middleware/PersistStache.php(22): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#42 [internal function]: Statamic\Http\Middleware\PersistStache->handle(Object(Illuminate\Http\Request), Object(Closure))
#43 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#44 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(50): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#45 [internal function]: Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle(Object(Illuminate\Http\Request), Object(Closure))
#46 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#47 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#48 [internal function]: Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#49 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#50 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(62): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#51 [internal function]: Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#52 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#53 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#54 [internal function]: Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#55 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#56 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(59): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#57 [internal function]: Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#58 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#59 /home/trinitylife/public_html/statamic/core/Http/Middleware/Glide.php(43): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#60 [internal function]: Statamic\Http\Middleware\Glide->handle(Object(Illuminate\Http\Request), Object(Closure))
#61 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#62 /home/trinitylife/public_html/statamic/core/StaticCaching/Middleware/Retrieve.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#63 [internal function]: Statamic\StaticCaching\Middleware\Retrieve->handle(Object(Illuminate\Http\Request), Object(Closure))
#64 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#65 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(44): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#66 [internal function]: Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#67 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#68 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#69 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#70 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(122): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#71 /home/trinitylife/public_html/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(87): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#72 /home/trinitylife/public_html/index.php(156): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#73 {main}
```

**Comment by castirondesign** [comment]
I'm experiencing something similar where I can't save globals without clearing the cache. Soon as I save one I can't save it or any others until I clear the cache again. 500 error on /cp/publish

This is only on 2.1.9, works fine if I downgrade to 2.1.8

**Comment by jcohlmeyer** [comment]
I have not tested this but I suspect that deleting a second asset without navigation in the cp media browser will result in the file being deleted from the file system but the yml not being updated to reflect the change.

This is the original issue that caused me to look into this.

**Comment by jasonvarga** [comment]
This is a confirmed bug if you have turned off stache updates on every
request.

**Comment by jcohlmeyer** [comment]
@jasonvarga I see this issue even if I have turned on stache updates on every request.

**Comment by jasonvarga** [comment]
Can you please post the fieldset for the entry you're updating, and point out the asset field in question.

**Comment by jasonvarga** [comment]
We fixed some related stuff in 2.1.11, so this might have been fixed in the process. Let me know how it works out after you update.

**Comment by jackmcdade** [comment]
Try now in 2.1.12. Gonna close this, we were able to replicate and it's fixed, but feel free to reopen if you find a new way to trigger an issue!

**Comment by jcohlmeyer** [comment]
Sorry for the delay in responding I was on vacation for a few weeks.

@jackmcdade I can confirm this issue is fixed for me as of 2.1.12.

**Comment by jackmcdade** [comment]
Great!
